pycue/opencue/cuelogging.py

```diff
@@ -0,0 +1,166 @@
+#  Copyright Contributors to the OpenCue Project
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+
+"""Module for reading and writing log files"""
+
+import logging
+import os
+import platform
+import datetime
+import time
+
+log = logging.getLogger(__name__)
+
+MODE_READ = 0
+MODE_WRITE = 1
+
+class CueLogger(object):
+    """Class to abstract file logging, this class tries to act as a file object"""
+    filepath = None
+    fd = None
+    mode = MODE_READ  # Default in read mode
+
+    def __init__(self, filepath, mode, maxLogFiles=1):
```

Comment:
This function is too big for an init, I'm suggesting bellow some sections to be extracted into smaller functions to make the logic easier to read:

Reply:
Have split it into 2 separate classes, so should not be a concern anymore.